iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -138,6 +148,17 @@ public void initialize(@Nullable Configuration configuration, Properties serDePr
     }
   }
 
+  private void createTableForCTAS(Configuration configuration, Properties serDeProperties) {
+    serDeProperties.setProperty(TableProperties.ENGINE_HIVE_ENABLED, "true");
+    serDeProperties.setProperty(InputFormatConfig.TABLE_SCHEMA, SchemaParser.toJson(tableSchema));
+    Catalogs.createTable(configuration, serDeProperties);
+    // set these in the global conf so that we can rollback the table in the lifecycle hook in case of failures
```

Comment:
A good candidate to put something into `QueryInfo`, or somewhere else

Reply:
I think so too. I'll table this topic for now though, and will create a follow-up patch to put the job commit info (as discussed in the write commit PR) and this ctas info into this object.